ui/apps/platform/src/Containers/Policies/Wizard/Step2/PolicyBehaviorForm.tsx

```diff
@@ -60,7 +82,28 @@ function PolicyBehaviorForm() {
                 values.enforcementActions
             );
         }
-        setValues(changedValues);
+        return changedValues;
+    }
+
+    function onConfirmChangeLifecycle(
+        lifecycleStage: LifecycleStage | undefined,
+        isChecked: boolean | undefined
+    ) {
+        // type guard, because TS is a cruel master
+        if (lifecycleStage) {
+            // first, update the lifecycles
+            const newValues = getLifeCyclesUpdates(lifecycleStage, !!isChecked);
+
+            // second, clear the policy criteria
+            const clearedCriteria = initialPolicy.policySections;
```

Comment:
Out of an abundance of caution, what do you think about `cloneDeep(initialPolicy.policySections)` here, as a less verbose way to achieve the same objective as a snippet from about 50 lines below in the unchanged lines of this pull request:

```js
        setFieldValue(
            'policySections',
            [
                {
                    sectionName: 'Policy Section 1',
                    policyGroups: [],
                },
            ],
            false
        );
    }
```

Although we trust formik to cloneDeep initialPolicy for values, if any of our event handlers ever made an impure update to policySections, we would overwrite the factory default.

Reply:
Since I've refactored that that initialPolicy struct out, it seems I should replace that other reset instance with the cloneDeep, too.

@alwayshooin , You are the other Policies expert on the team. Can you see any reason not to leverage cloneDeep in both places?

Reply:
hmm the only concern i can think of is performance issues that might arise, particularly in formik expensive areas like the step 3 section that have a lot of formik `Field`s, but i don't see why we shouldn't try and amend if it turns out to be too pricey

Reply:
Because these are rare edge cases, correctness is important and performance is not a problem.

```js
        const clearedCriteria = cloneDeep(initialPolicy.policySections);
```

```js
        // clear policy sections to prevent non-runtime criteria from being sent to BE
        setFieldValue('policySections', cloneDeep(initialPolicy.policySections));
```